src/main/java/ru/testtask/service/UserService.java

```diff
@@ -17,24 +19,30 @@
     @Autowired
     private UserRepo userRepo;
 
+    @SneakyThrows
     @Override
-    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
-        User user = userRepo.findByUsername(username);
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException{
+            User user = userRepo.findByUsername(username);
 
-        if (user == null){
-            throw new UsernameNotFoundException("User does not exists");
-        }
+            if (user == null) {
+                throw new UsernameNotFoundException(String.format("User with username %s does not exists", username));
+            }
 
-        return user;
+            return user;
     }
 
     public User getUserByUsername(String username){
         return userRepo.findByUsername(username);
     }
 
-    public String getCurrentUserId(){
-        org.springframework.security.core.userdetails.User user = (org.springframework.security.core.userdetails.User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
-        return getUserByUsername(user.getUsername()).getId();
+    public String getCurrentUserId() throws WrongMethodUseException{
+        try {
+            org.springframework.security.core.userdetails.User user = (org.springframework.security.core.userdetails.User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```

Comment:
It is better to check `SecurityContextHolder.getContext().getAuthentication()` value. If value is null you can throw exception with message 
`Method can't be used from this thread. `

Now the behaviour